### PR TITLE
[Core][Node Labels 2/n] Add labels param in python ray.init() to setting node labels

### PR DESCRIPTION
**Pull request by larrylian**
## Why are these changes needed?
Add labels param in python ray.init() to setting node labels
user cases:
```
  labels = {"gpu_type": "A100", "region": "us"}
  ray.init(labels=labels)
  assert ray.nodes()[0]["Labels"] == labels
  ray.shutdown()
  ray.init(labels={})
  assert ray.nodes()[0]["Labels"] == {}
```

## Related issue number
Enhancing node affinity scheduling feature through node labels https://github.com/ray-project/ray/issues/34894
> (P1)Finalize the format for setting node labels in the ray.init() interface of Python worker.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/_private/worker.py
Comment (jjyao) [inline python/ray/_private/worker.py:1308]:
```suggestion
        labels: [Experimental] The key-value labels of the node.
```

**Review by jjyao — APPROVED** [review]

File: python/ray/_private/worker.py
Comment (larrylian) [reply python/ray/_private/worker.py]: done

File: python/ray/tests/test_node_labels.py
Comment (jiwq) [inline python/ray/tests/test_node_labels.py:42]: If enable the node label feature, should all the node must add label?

File: python/ray/tests/test_node_labels.py
Comment (jiwq) [inline python/ray/tests/test_node_labels.py:43]: If enable the node label feature, should all the node must add label?

Comment (jiwq) [inline python/ray/tests/test_node_labels.py:43]: If enable the node label feature, should all the node must add label?

Comment (jjyao) [reply python/ray/tests/test_node_labels.py:43]: labels are optional for nodes.

File: python/ray/_private/worker.py
Comment (pcmoritz) [inline python/ray/_private/worker.py:1206]: Since these are kwargs and the ordering doesn't matter, let's put this next to `resources`, that will logically be the best place to put it :)

**Review by pcmoritz — APPROVED** [review]
Please address the comment before merging it :)

**Comment by larrylian** [comment]
> Please address the comment before merging it :)

I went on vacation last week, and I have fixed now, and I am waiting for the CI to be merge.

**Pull request merged**
